Owin.Security.Providers/Asana/AsanaAuthenticationOptions.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.Owin;
+using Microsoft.Owin.Security;
+
+namespace Owin.Security.Providers.Asana
+{
+    public class AsanaAuthenticationOptions : AuthenticationOptions
+    {
+        public class AsanaAuthenticationEndpoints
+        {
+            /// <summary>
+            /// Endpoint which is used to redirect users to request Asana access
+            /// </summary>
+            /// <remarks>
+            /// Defaults to https://app.asana.com/-/oauth_authorize
+            /// </remarks>
+            public string AuthorizationEndpoint { get; set; }
+
+            /// <summary>
+            /// Endpoint which is used to exchange code for access token
+            /// </summary>
+            /// <remarks>
+            /// Defaults to https://app.asana.com/-/oauth_token
+            /// </remarks>
+            public string TokenEndpoint { get; set; }
+
+            /// <summary>
+            /// Endpoint which is used to obtain user information after authentication
+            /// </summary>
+            /// <remarks>
+            /// Defaults to https://asana.com/1/OAuthGetRequestToken
+            /// </remarks>
+            public string UserInfoEndpoint { get; set; }
```

Comment:
The user info is returned in the initial authentication calls and this endpoint does not apply.  'UserInfoEndpoint' maybe should be removed to avoid confusion.